mobify/platform-scaffold
# Add push messaging

Adds push messaging to the scaffold. Note that the tests are failing since they can't access the private SDK repository.

 **Linked PRs**: https://github.com/mobify/progressive-web-sdk/pull/646

## Changes
- Add the push messaging controller and ~~soft-ask~~ ask to the app behind the `messagingEnabled` flag
- Update the messaging client init in loader to add the initPromise to `window` early
- Update components SCSS file with push messaging styles
- Add a page count increment action to `template.jsx`
- Add push messaging reducer to the store
- Updates `merlinspotions.yaml` web push configuration with a HTTPS welcome message URL

## TODO
- [x] Revert change to `package.json` after https://github.com/mobify/progressive-web-sdk/pull/646 is merged

## How to test-drive this PR
- Check out the branch associated with https://github.com/mobify/progressive-web-sdk/pull/646, i.e. `progressive-web-sdk#push-messaging` and run `npm install` for good measure
- Within `progressive-web-sdk`, run `npm run dev:build` to build the `dist/` folder, then run `npm link`
- Now, back in the scaffold, run `npm link progressive-web-sdk`
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Note that the "ask" is shown on the first page load. Check out the console to see what's happening and try accepting or dismissing the ask and system-ask and it should behave as you'd expect
